crates/krator-derive/src/lib.rs

```diff
@@ -1,96 +1,104 @@
 }
 
-fn parse_as_transition_attr(attr: Attribute) -> Option<Transitions> {
-    if let Some(id) = attr.path.get_ident() {
-        if id == ATTRIBUTE_NAME {
-            attr.parse_args::<Transitions>().ok()
-        } else {
-            None
-        }
-    } else {
-        None
-    }
-}
-
-fn generate_impl(transitions: Transitions, generics: Generics, name: Ident) -> TokenStream {
-    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
-    let mut token_stream = TokenStream::new();
-
-    for transition_type in transitions.all.into_iter() {
-        let expanded = quote! {
-            #[automatically_derived]
-            impl#impl_generics krator::TransitionTo<#transition_type> for #name#ty_generics #where_clause {}
-        };
-        token_stream.extend(TokenStream::from(expanded));
-    }
+#[cfg(feature = "admission-webhook")]
+mod admission;
 
-    token_stream
+#[cfg(feature = "admission-webhook")]
+#[proc_macro_derive(AdmissionWebhook)]
+///
+/// provides functions for creating necessary resources for running a admission webhook:
+///
+/// - `admission_webhook_resources(namespace: &str) -> (Service, Secret, MutatingWebhookConfiguration)`: Convenience function that returns all Kubernetes
+///   resources necessary to configure an admission webhook service. It expects a deployed pod with label app having the value returned by admission_webhook_service_name().
+/// - `admission_webhook_secret_name() -> String`: Returns the name of the admission webhook secret that will get created with `admission_webhook_secret()`
+/// - `admission_webhook_service_app_selector() -> String`: Returns the selector value for the label app that the service created with
+///    `admission_webhook_service()` uses for selecting the pods that serve the admission webhook
+/// - `admission_webhook_service_name() -> String`:  Returns the name of the admission webhook service that will get
+///    created with `admission_webhook_service()`
+/// - `admission_webhook_configuration_name() -> String`:Returns the name of the admission webhook configuration that will get
+///    created with `admission_webhook_configuration()`
+/// - `admission_webhook_secret(namespace: &str) -> Secret`: Creates a Kubernetes secret of type `tls` that contains a
+///    certificate and a private key and can be used for the admission webhook service
+/// - `admission_webhook_service(namespace: &str) -> Service`: Returns a service that forwards to pods where label `app`
+///    has the value returned by the function `admission_webhook_service_app_selector()`. It exposes port `443`
+///    (necessary for webhooks) and listens to the pod's port `8443`
+/// - `admission_webhook_configuration(service: Service, secret: Secret) -> Result<MutatingWebhookConfiguration>`:  Creates a MutatingWebhookConfiguration using
+///    the certificate from the given service and the service of the given service as configuration
+///
+/// Example
+/// ```
```

Comment:
This is a great example! The one problem (like in a previous comment) that this code cannot depend on `krator` or we get a weird circular dependency issue. If we can get it so that isn't needed in `dev-dependencies`, this will work fine. If it does need a dependency on `krator`, we'll want to move this example there

Reply:
is it really a problem in `dev-dependencies` as well? I got problems when having circular dependencies in `dependencies` but no complains in `dev-dependencies` ... but that's perhaps just a cheap `works for me` ;)

Reply:
I think we are ok for now. I'll do some double checking later